src/platform/plugins/shared/discover/public/application/main/state_management/redux/types.ts

```diff
@@ -58,13 +34,14 @@ export interface TabStateGlobalState {
 }
 
 export interface TabState extends TabItem {
-  // Initial app and global state for the tab (provided before the tab is initialized).
+  // Initial state for the tab (provided before the tab is initialized).
+  initialInternalState?: {
+    serializedSearchSource?: SerializedSearchSourceFields;
+  };
```

Comment:
This allows us to locally persist non-URL tab state which is necessary for ad hoc data views in uninitialized tabs. This can also be used for `visContext` too. Not sure on the "internal state" naming though, maybe there's a clearer alternative.